### PR TITLE
chore/ci: move profiling to benchmark.js

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

The 2nd part of https://github.com/hexojs/hexo/issues/4437.

After the PR, GitHub Action will perform both benchmark & profiling (without deploying flamegraph), while Travis CI will only perform profiling and deploy flamegraph.

## How to test

Just see if Travis CI works.

An example deployed flamegraph: http://ceaa0cee07fc8faea7958c70fdc1208e55d20070-14-hexo.surge.sh/flamegraph.html

## Screenshots

Not applicable.

## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

File: test/benchmark.js [hunk 1]
```diff
@@ -21,17 +21,37 @@ const isWin32 = require('os').platform() === 'win32';
 const npmScript = isWin32 ? 'npm.cmd' : 'npm';
 
 const testDir = resolve('.tmp-hexo-theme-unit-test');
+const zeroEksDir = process.env.TRAVIS_BUILD_DIR
+  ? join(process.env.TRAVIS_BUILD_DIR, '0x')
+  : resolve(testDir, '0x');
```
Comment (SukkaW) [inline test/benchmark.js:26]: To be compatible with current Travis CI configuration. Should be removed in the future.

File: test/benchmark.js [hunk 2]
```diff
@@ -21,17 +21,37 @@ const isWin32 = require('os').platform() === 'win32';
...
 
+const zeroEks = require('0x');
+
+let isProfiling = process.argv.join(' ').includes('--profiling');
+let isBenchmark = process.argv.join(' ').includes('--benchmark');
+
+if (!isProfiling && !isBenchmark) {
+  isProfiling = true;
+  isBenchmark = true;
+}
```
Comment (SukkaW) [inline test/benchmark.js:37]: Run both profiling and benchmark when neither `--profiling` nor `--benchmark` exists.

File: test/benchmark.js [hunk 3]
```diff
@@ -129,3 +149,24 @@ async function init() {
...
+
+async function profiling() {
+  // Clean up 0x dir before profiling
+  if (await exists(zeroEksDir)) await rmdir(zeroEksDir);
+
+  const zeroEksOpts = {
+    argv: [hexoBin, 'g', '--cwd', testDir],
+    workingDir: '.', // A workaround for https://github.com/davidmarkclements/0x/issues/228
+    outputDir: zeroEksDir,
+    title: 'Hexo Flamegraph'
```
Comment (SukkaW) [inline test/benchmark.js:161]: The `title` of generated flamegraph.

**Comment by curbengh** [comment]
It would be nice to have `"echo "Flamegraph will be deployed to: https://${TRAVIS_COMMIT}-${TRAVIS_NODE_VERSION}-hexo.surge.sh/flamegraph.html"` in the build log.

Is travis going to be replaced eventually? There is a surge.sh [action](https://github.com/marketplace/actions/publish-to-surge-sh).

**Review by Himself65 — APPROVED** [review]

**Comment by SukkaW** [comment]
@curbengh 

> Is travis going to be replaced eventually?

Yes.

> There is a surge.sh [action](https://github.com/marketplace/actions/publish-to-surge-sh).

I have no permission to modify the `SECRET` settings of hexo repo, thus I can not bring my own surge account to GitHub Action (Unlike on Travis CI I am using my own account). But we will find out the way (or uses other deployment).

**Comment by curbengh** [comment]
I can add `secret` for you.

**Comment by SukkaW** [comment]
> I can add `secret` for you.

@curbengh If so, you can add your own account as well. `surge.sh` just doesn't care about our usage:

https://github.com/hexojs/hexo/pull/4000#issuecomment-568662111
https://github.com/hexojs/hexo/pull/4000#issuecomment-574628327

**Comment by curbengh** [comment]
Added `SURGE_LOGIN` and `SURGE_TOKEN` secrets.

**Pull request merged**